lang/ir/src/ir/item_impl/callable.rs

```diff
@@ -362,12 +363,17 @@ pub(super) fn ensure_callable_invariants(
     method_item: &syn::ImplItemMethod,
     kind: CallableKind,
 ) -> Result<(), syn::Error> {
-    if !matches!(method_item.vis, syn::Visibility::Public(_) | syn::Visibility::Inherited)
-    {
-        return Err(format_err_spanned!(
-            method_item.vis,
+    let bad_visibility = match &method_item.vis {
```

Comment:
Can use `utils::ensure_pub_visibility`

Reply:
Unfortunately we cannot make use of that API since the callables also allow inherited visibility.